ee/billing/quota_limiting.py

```diff
@@ -89,6 +92,14 @@ def org_quota_limited_until(organization: Organization, resource: QuotaResource)
     if is_quota_limited and organization.never_drop_data:
         return None
 
+    if is_quota_limited and posthoganalytics.feature_enabled(QUOTA_LIMIT_DATA_RETENTION_FLAG, organization.id):
```

Comment:
Assuming this is a group flag, you need the groups parameter here, along with any potential group properties you'll use to target orgs so this can be evaluated locally quickly